### PR TITLE
Update flake8 config

### PR DESCRIPTION
**Pull request by srittau**
* Improve documentation on why which attributes are ignores.
* Add links to issues where appropriate.
* Ignore E501 (handled by black) in script files and remove max-line-length.
* Limit Python 2 builtins to those actually used.
* Don't ignore W503, which was removed from the list of default rules.

### CODE REVIEW

**Comment by AlexWaygood** [comment]
black actually declines to split string literals into multiple lines, whereas flake8 will still raise an error if a line is too long — I've found flake8's line-length checking useful for PRs made on `check_new_syntax.py`

**Comment by srittau** [comment]
There are more incompatibilities. For example, black sometimes keeps a `# type: ignore` beyond the line limit. I prefer not to worry about this too much.

File: .flake8 [hunk 1]
```diff
@@ -1,26 +1,32 @@
...
-#       E501 line too long
 
 [flake8]
 per-file-ignores =
-  *.py: E203, W503
+  *.py: E203, E501
   *.pyi: E301, E302, E305, E501, E701, E741, F401, F403, F405, F822
   # Since typing.pyi defines "overload" this is not recognized by flake8 as typing.overload.
   # Unfortunately, flake8 does not allow to "noqa" just a specific error inside the file itself.
+  # https://github.com/PyCQA/flake8/issues/1079
```
Comment (Akuli) [inline .flake8:26]: This seems to be no longer true. The linked issue is closed.

Comment (srittau) [reply .flake8:26]: I tested it, it's still not possible. I believe the issue was closed, because the maintainers have no intention of implementing this.

**Pull request merged**